### ISSUE TITLE
Magento 2.3.4 compatibility (magento/framework ~102.0.4)

### ISSUE THREAD

**Issue by rutgerrademaker** [buckaroo-it/Magento2#4]
When upgrading magento to 2.3.4 (Commerce) I ran into the issue that this package is not compatible yet.

The issue is that `magento/framework:102.0.4` is not allowed as a requirement.

Could you please verify that  102.0.4 can safely be added in the package's allowed requirements?

**Comment by vladislav-padalka-hys** [comment]
Hi Rutger !

It's natural that we will test and make compatible our plugin with v2.3.4 as soon as possible, but can you please clarify the way you did an update to 2.3.4 ?
Because even it was announced some time ago, but currently we see no such version available for public download or composer installation.

**Comment by rutgerrademaker** [comment]
@vladislav-padalka-hys Ahh.. the company I work for has access to the "Pre-release" (which comes 2 weeks before GA)
So, we can already install it via composer. Not sure if it is available in the Open Source version too.
I'd suggest just to try to install it via composer (that's what I did)

**Comment by Buckaroo-Rens** [comment]
@rutgerrademaker Thank you for your feedback. We allready tried true composer without any results. Since we are still awaiting our partner account there is no other option than waiting for the full release. Keep you posted.

**Comment by ericclaeren** [comment]
Why did you close this issue? This isn't resolved yet.
 
The 2.3.4 commerce version is available, when do you expect to get your partner account?
Because this is holding back our update to 2.3.4.

**Comment by igorwulff** [comment]
Hi @Buckaroo-Rens the minor update includes security fixes as well, so we want to be up to date asap and be able to start testing before the pre-release is released. Can some kind of beta or release candidate be provided? You only have to alter the restriction in the composer.json in this version.

**Comment by ericclaeren** [comment]
@igorwulff You can use 2.3.3-p1 for the security fixes. 

We are upgrading from 2.3.2-p2, so instead of upgrading to 2.3.4 we now need to go to 2.3.3 and this means we have to test the entire site for the 2.3.3-p1 upgrade and a second time when upgrading to 2.3.4, which is a costly process, for us and the customer. So this is the reason @Buckaroo-Rens why we rather would update to 2.3.4 directly.

You are correct that 2.3.4 is a minor version, but Magento has introduced a lot of fixes and new features. The changelog is pretty huge for this 'minor'

https://devdocs-beta.magento.com/guides/v2.3/release-notes/release-notes-2-3-4-commerce.html

**Comment by vladislav-padalka-hys** [comment]
Hi @igorwulff !

We've made https://github.com/buckaroo-it/Magento2/releases/tag/v1.16.1-RC1 pre-release with "magento/framework:102.0.4" requirement you've requested. 

So please try to update with "buckaroo/magento2:1.16.1-RC1" and let us know if help.

**Comment by ericclaeren** [comment]
Thanks @vladislav-padalka-hys 

Going to test it, will provide feedback as soon as we have tested the upgrade.

**Comment by Buckaroo-Rens** [comment]
> Going to test it, will provide feedback as soon as we have tested the upgrade.

Hi Eric, can you give us any update regarding the tests.

**Comment by ericclaeren** [comment]
> > Going to test it, will provide feedback as soon as we have tested the upgrade.
> 
> Hi Eric, can you give us any update regarding the tests.

No, sorry, the update is crashing on yet unknown code which is not related to Buckaroo, need to fix this before continuing testing.

**Comment by ericclaeren** [comment]
@Buckaroo-Rens 

Tested the latest release: 1.16.1 and everything in test mode went fine. 👍 

The only issue I ran in was a configuration issue where the merchant key was overridden by store view, which I didn't directly see and the value seemed not to be cleared by the updateData or reset to the default value (website).

Thanks!

**Comment by Buckaroo-Rens** [comment]
> The only issue I ran in was a configuration issue where the merchant key was overridden by store view, which I didn't directly see and the value seemed not to be cleared by the updateData or reset to the default value (website).

Can you confirm that you were updating from v1.14.0 or lower? Since in v1.14.0 was a security update were you needed to fill in the merchant/secrect key again;

> **Your action is required!**
> You have to fill in your Merchant and Secret key(s) again in the configuration to continue using Buckaroo. With this version we have improved the overall security, but reentering your keys is needed.

**Comment by ericclaeren** [comment]
Hi @Buckaroo-Rens.

Yes we were updating from a lower version. But I expected the update script to remove all current keys, instead the overriden store view key seems not to have been deleted.

**Comment by Buckaroo-Rens** [comment]
> Yes we were updating from a lower version. But I expected the update script to remove all current keys, instead the overriden store view key seems not to have been deleted.

Thank you for the information. We will test this later this sprint and share te results with you.

**Comment by vladislav-padalka-hys** [comment]
Hi Eric ! 
We've tried to reproduce the issue you have: installed version of 1.13.1 (before merchant/security keys change), entered merchant keys for all levels of configuration, then upgraded to 1.14.0 (where is merchant/security keys change) and as result merchant keys values were erased (replaced with empty strings) in DB.
It's natural that in UI they still be shown as asterix , but in DB they are empty strings.

If you will provide some more details about your case then we can try to reproduce the issue, but assuming SQL query we have in code (it simply updates by empty value all entries in core_config_data table with 'account/merchant_key' paths) and tests we did it looks like it may be something on your side.
